m3/src/reporter_builder.cc

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "m3/reporter_builder.h"
+
+namespace tally {
+
+namespace m3 {
+
+const uint32_t ReporterBuilder::DEFAULT_MAX_QUEUE_SIZE = 1024;
+const uint16_t ReporterBuilder::DEFAULT_MAX_PACKET_SIZE = 1440;
+const uint16_t ReporterBuilder::DEFAULT_PORT = 9052;
+const char ReporterBuilder::DEFAULT_HOST[] = "127.0.0.1";
+const std::map<std::string, std::string> ReporterBuilder::DEFAULT_COMMON_TAGS =
+    {};
+
+ReporterBuilder::ReporterBuilder()
+    : host_(DEFAULT_HOST),
+      port_(DEFAULT_PORT),
+      common_tags_(DEFAULT_COMMON_TAGS),
+      max_queue_size_(DEFAULT_MAX_QUEUE_SIZE),
+      max_packet_size_(DEFAULT_MAX_PACKET_SIZE) {}
+
+ReporterBuilder &ReporterBuilder::host(const std::string &host) {
```

Comment:
Please use clang-format (and define a style for yourself).

Reply:
OK just another anomaly of Google style. Have never seen reference separated from type by a space.